src/UIComponents/Statement.jsx

```diff
@@ -40,65 +99,42 @@ class Statement extends Component {
+    const predictionStatementParts = predictionStatement.split(regex).filter((part) => part);
+    // Swap the OUTPUT and INPUTS placeholders with React components.
+    const predictionStatementComponents = predictionStatementParts.map((part, index) => {
+      if (part === OUTPUT_KEY) {
+        return this.labelHTML(labelColumn, currentPanel);
+      } else if (part === INPUTS_KEY) {
+        return this.selectedFeaturesHTML(selectedFeatures, currentPanel);
+      } else {
+        return part;
+      }
+    }).map((part, index) => {
+      //React wants sibling elements in a list to have a unique "key".
+      return (<span key={index}>{part}</span>);
+    });
+
     return (
       <div
         style={smallFont ? styles.statementSmall : styles.statement}
         id="statement"
       >
-        Predict{" "}
-        <div style={styles.statementLabel}>
-          {labelColumn || "____"}
-          {currentPanel === "dataDisplayLabel" && labelColumn && (
-            <div
-              id="uitest-remove-statement-label"
-              onClick={() => this.removeLabel()}
-              onKeyDown={() => this.removeLabel()}
-              style={styles.statementDeleteIcon}
-              role="button"
-              tabIndex={0}
-            >
-              <div style={styles.statementDeleteCircle} />
-              <div style={styles.statementDeleteX}>
-                <FontAwesomeIcon icon={faTimesCircle} />
-              </div>
-            </div>
-          )}
-        </div>
-        {currentPanel !== "dataDisplayLabel" && (
-          <span>
-            {" "}
-            based on{" "}
-            {selectedFeatures.map((selectedFeature, index) => {
-              return (
-                <span key={index}>
-                  <div style={styles.statementFeature}>
-                    {selectedFeature}
-                    {currentPanel === "dataDisplayFeatures" && (
-                      <div
-                        id="uitest-remove-statement-feature"
-                        onClick={() => this.removeFeature(selectedFeature)}
-                        onKeyDown={() => this.removeFeature(selectedFeature)}
-                        style={styles.statementDeleteIcon}
-                        role="button"
-                        tabIndex={0}
-                      >
-                        <div style={styles.statementDeleteCircle} />
-                        <div style={styles.statementDeleteX}>
-                          <FontAwesomeIcon icon={faTimesCircle} />
-                        </div>
-                      </div>
-                    )}
-                  </div>
-                  {index < selectedFeatures.length - 1 && ", "}
-                </span>
-              );
-            })}
-            {selectedFeatures.length === 0 && (
-              <span style={styles.statementFeature}>____</span>
-            )}
-            .
-          </span>
-        )}
+        {predictionStatementComponents}
```

Comment:
Apologies for the minor nits.  This is a really fun approach!  

It looks like it works for all permutations, but have you had the chance to play around with different orderings, imagining what different translations might do, and convinced yourself it works for all of them?

Reply:
It seems like it will work for most of them. There only seems to be one minor nitpick if the {inputs} and {output} are right next to each other because the `X` to remove the selected feature overlaps a little with the orange box.
![image](https://user-images.githubusercontent.com/1372238/199850319-5a9eb5e6-b128-431b-aa06-15f9cad68820.png)
But we don't know if any languages will actually have this layout. I think its safe to ship this, get translations back, then re-evaluate.

Reply:
I also did a manual test with Arabic, which is a right-to-left language, and everything looked like it rendered in the correct order (at least compared to google translate).